ast.go

```diff
@@ -28,13 +28,23 @@ import (
 type Serializer interface {
 	// Serialize writes the statement/expression to the io.Writer. If an
 	// error is returned the io.Writer may contain partial output.
-	Serialize(w io.Writer) error
+	Serialize(w io.Writer, interpolate bool) error
```

Comment:
Would that mean `Serialize` would take in a `sqlWriter`? Or would you do a type switch in each Val's `Serialize` to determine if it's a `sqlWriter` or generic `io.Writer`?